src/server/scripts/Northrend/Naxxramas/instance_naxxramas.cpp

```diff
@@ -431,6 +453,18 @@ class instance_naxxramas : public InstanceMapScript
         
         bool SetBossState(uint32 bossId, EncounterState state) override
         {
+            // pull all the trash if not killed
+            if (bossId == BOSS_PATCHWERK)
+                if (state == IN_PROGRESS)
+                    if (Creature* patch = instance->GetCreature(_patchwerkGUID))
+                        if (!PatchwerkRoomTrash.empty())
```

Comment:
why not using a single IF with (condition1 && condition2 && ...)

Reply:
You cant make the pointer and a check on the same if, but if you talk about bossid and state sure, i will change it.

Reply:
yeah so many ifs, will merge after fix

Reply:
if (bossId == BOSS_PATCHWERK) &&  (state == IN_PROGRESS) && (Creature* patch = instance->GetCreature(_patchwerkGUID)) && if (!PatchwerkRoomTrash.empty())

Would this work? (looks bad too though)